MetaMask/metamask-extension
# Feat/swaps link buy gas

## Explanation



This PR resolves [MMS-525](https://consensyssoftware.atlassian.net/browse/MMS-525)

> Problem

> When a user does not have enough gas to complete a transaction, we show them an error message & they are at a dead end.

> Proposed Solution

> We should offer a link to on-ramps so that they can top-up their balance, allowing them to complete swaps. On mobile, this is a hyperlink at the end of the message, saying “Buy more ETH”. We would update “ETH” to match the native token symbol if the user is not on eth mainnet.

1. Adds a link to buy ETH (or whatever native gas token is needed for the network) when the gas is insufficient to complete the swap.

## Screenshots/Screencaps



### Before


![Screenshot 2023-07-27 at 3 20 00 PM](https://github.com/MetaMask/metamask-extension/assets/139582705/3c5642d9-4b9c-49f5-913b-19ed7839831e)

### After


![Screenshot 2023-07-27 at 3 45 33 PM](https://github.com/MetaMask/metamask-extension/assets/139582705/11487d10-d81b-450c-adab-d9d0ab9cb656)



## Manual Testing Steps



1. Go to Swaps
2. Select the gas token for the current network as the From token
3. Enter in an amount greater or equal to your current balance for From token
4. Observe the link to "Buy ETH"
5. Click the link
6. A new tab should open showing the Portfolio and its Buy page

Make sure to test both with Smart Transactions On and Off

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
